include/seqan3/io/sequence_file/format_sam.hpp

```diff
@@ -0,0 +1,279 @@
+// Copyright (c) 2006-2019, Knut Reinert & Freie Universität Berlin
+// Copyright (c) 2016-2019, Knut Reinert & MPI für molekulare Genetik
+// This file may be used, modified and/or redistributed under the terms of the 3-clause BSD-License
+// shipped with this file and also available at: https://github.com/seqan/seqan3/blob/master/LICENSE
+// -----------------------------------------------------------------------------------------------------
+
+/*!\file
+ * \brief Provides the seqan3::sequence_file_format_sam class.
+ * \author Mitra Darvish <mitra.darvish AT fu-berlin.de>
+ */
+
+#pragma once
+
+#include <iterator>
+#include <string>
+#include <string_view>
+#include <vector>
+
+#include <range/v3/algorithm/copy.hpp>
+#include <range/v3/utility/iterator.hpp>
+
+#include <seqan3/alphabet/nucleotide/dna5.hpp>
+#include <seqan3/core/metafunction/range.hpp>
+#include <seqan3/io/detail/misc.hpp>
+#include <seqan3/io/sequence_file/input_options.hpp>
+#include <seqan3/io/sequence_file/output_options.hpp>
+#include <seqan3/io/stream/parse_condition.hpp>
+#include <seqan3/range/detail/misc.hpp>
+#include <seqan3/range/view/char_to.hpp>
+#include <seqan3/range/view/to_char.hpp>
+#include <seqan3/range/view/take.hpp>
+#include <seqan3/range/view/take_exactly.hpp>
+#include <seqan3/range/view/take_line.hpp>
+#include <seqan3/range/view/take_until.hpp>
+#include <seqan3/std/ranges>
+#include <seqan3/std/view/subrange.hpp>
+#include <seqan3/std/view/transform.hpp>
+
+namespace seqan3
+{
+  /*!\brief       The SAM format used as sequence file.
+   * \implements  sequence_file_format_concept
+   * \ingroup     sequence
+   *
+   * \details
+   *
+   * ### Introduction
+   *
+   * The SAM format is commonly used to store pairwise alignment information between a query sequence and its
+   * reference sequence, e.g. a read mapping result. As it provides information on the **id**, **sequence** and
+   * **quality** of each query, it can additionally be treated as a sequence file.
+   * See the [article on Wikipedia](https://en.wikipedia.org/wiki/SAM_(file_format)) or the 
+   * [technical specification] (https://samtools.github.io/hts-specs/SAMv1.pdf) for an in-depth description of
+   * the format.
+   *
+   * ### Fields
+   *
+   * The SAM format provides the fields seqan3::field::SEQ, seqan3::field::ID and seqan3::field::SEQ_QUAL.
+   * All fields are allowed to be empty when writing.
+   *
+   * ### Implementation notes
+   *
+   * This implementation supports the following optional features of the format:
+   *
+   * line breaks and/or other whitespace characters in any part of the sequence (only when reading!), optional tags
+   * are going to be ignored (when reading)
+   *
+   */
+class sequence_file_format_sam
+{
+public:
+    /*!\name Constructors, destructor and assignment
+     * \{
+     */
+     //!\brief Default constructor is explicitly deleted, you need to give a stream or file name.
+    sequence_file_format_sam() = default;
```

Comment:
the default constructor is defaulted, but the documentation says it's deleted...